test/serializers/associations_test.rb

```diff
@@ -195,6 +195,39 @@ def test_belongs_to_allows_id_overwriting
         assert_equal expected, actual
       end
 
+      class BlogWithCustomType < Blog
+        attributes :id
+      end
+      class BlogWithCustomTypeSerializer < ActiveModel::Serializer
+        attributes :id
+        type 'custom-type'
+      end
+      class BelongsToBlogWithCustomTypeModel < ::Model
+        attributes :id, :title, :blog_with_custom_type_id
+        associations :blog_with_custom_type
+      end
+      class BelongsToBlogWithCustomTypeModelSerializer < ActiveModel::Serializer
+        type :posts
+        belongs_to :blog_with_custom_type
+      end
+
+      def test_belongs_to_allows_type_overwriting
```

Comment:
`test_belongs_to_doesnt_load_record` should also change, no, since it requires `belongs_to_uses_id_on_self` to be true, no?